helix-core/src/main/java/org/apache/helix/task/TaskAssignmentCalculator.java

```diff
@@ -121,4 +122,39 @@ public static String getQuotaType(WorkflowConfig workflowConfig, JobConfig jobCo
     }
     return deletedPartitions;
   }
+
+  /**
+   * Get all the partitions/tasks that belong to the non-targeted job.
+   * @param jobCfg the task configuration
+   * @param jobCtx the task context
+   * @return set of partition numbers
+   */
+  public Set<Integer> getAllTaskPartitionsDefault(JobConfig jobCfg, JobContext jobCtx) {
+    Map<String, TaskConfig> taskMap = jobCfg.getTaskConfigMap();
+    Map<String, Integer> taskIdMap = jobCtx.getTaskIdPartitionMap();
+    // Check if a gap exists in the context due to previouslys
+    // removed tasks. If yes, the missing pIDs should be considered
```

Comment:
This might cause problems if we want to debug for what happened to a removed task from the log. Ideally, we should avoid any overlap even with the removed task Id. However, I also understand that even we pick up the current max Id, there is no guarantee the next number has not been used before.
Let's keep the design.

One question, though, can we just use the task Id to replace task Partition Id completely? That will resolve all the problems, right?